### PR TITLE
Upgrade Checkstyle to 7.6

### PR DESCRIPTION
**Pull request by robertpainsi**
Mostly updating checks and adding new ones. I also reformatted the whole project to check for disagreements between Checkstyle and the Android Studio Formatter and I found one. Arrays and enums were definitely a problem. So I had to made a drastically change. No whitespaces within arrays and enums. ~`new int[] { 1, 2 }`~ → `new int[] {1, 2}`. I'd prefer it anyway because it's also the [default setting](https://github.com/Catrobat/Catroid/pull/2206/files#diff-f1045aea3015ed5f80704025e3ee300eL92) in Android Studio. For more details see last commit. I added a `Under discussion` label because of this.

I already closed all according Jira tickets by moving them to `Rejected` :laughing:
For review, I suggest to look at each commit on it's own. It'll be sooo much easier :wink:

### CODE REVIEW

File: catroid/config/checkstyle.xml [hunk 1]
```diff
@@ -172,31 +179,51 @@
...
-      <property name="tokens" value="DOT" />
+      <property name="tokens" value="DOT, AT" />
       <property name="option" value="nl" />
     </module>
     <module name="SeparatorWrap">
-      <property name="tokens" value="COMMA" />
+      <property name="tokens" value="COMMA, ELLIPSIS, ARRAY_DECLARATOR, RBRACK" />
       <property name="option" value="EOL" />
     </module>
+    <module name="SingleSpaceSeparator" />
```
Comment (robertpainsi) [inline catroid/config/checkstyle.xml]: …check and I'm really proud of it :smirk_cat:

**Comment by robertpainsi** [comment]
@Catrobat/catroid, please someone review asap. :kissing_smiling_eyes:
Since the @CatrobatBot will be down for some time, these changes are really important.

Everyone fine with changing the array style?

File: catroid/config/checkstyle.xml (hunk 1 above)
Comment (thmq) [inline catroid/config/checkstyle.xml]: 👍

**Comment by thmq** [comment]
I'm fine with it. How thoroughly do you need me to review this? 😆

**Comment by robertpainsi** [comment]
> How thoroughly do you need me to review this?

The first (mainly fixing wrong modifiers due to the upgrade) and last two commits (reformat and new array style) are the "heavy" commits. All other commits are mostly adapting or adding new checks without fixing anything. I also added a Checkstyle link to each commit so you can easily look up the new checks. Checkstyle and the Android Studio Autoformatter also agree on each other.

Since we have big code base already, a misconfigured or inappropriate check should already be highlighted by many changes. Besides these three heavy commits mentioned above, only https://github.com/Catrobat/Catroid/pull/2206/commits/0edc01c6f00f7bcb35a5853dec128a1df4ed5b66 has a couple of changes just because e[1-9] (no enumeration, `e` is still fine) or i*O*Exception (wrong camelcase) should be renamed. The last two (of the heavy) commits have been made by the autoformatter.

So to answer your question, not that much :wink:

**Comment by thmq** [comment]
ok to test

**Comment by thmq** [comment]
one checkstyle warning on the jenkins build 😆 ironic isn't it?

**Comment by robertpainsi** [comment]
> one checkstyle warning on the jenkins build :laughing: ironic isn't it?

:laughing:

The build is :green_heart: now. I fixed the new Checkstyle warning by rebasing and avoided another merge commit. I retested each commit and also reformatted the whole project in the last two commits, as done before. This scenario is one of the downsides of using a merge-based approach. Both branches are correct, however, the merge causes problems.

**Pull request merged**